apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -274,32 +407,169 @@ export function buildMachine(
           PAPER_IN_OUTPUT: {
             target: 'scanning',
           },
+          PAPER_JAM: {
+            target: 'jammed',
+          },
         },
       },
       scanning: {
-        invoke: {
-          id: 'scanAndSave',
-          src: (context) => {
-            return scanAndSave(context.driver);
+        invoke: [
+          {
+            id: 'scanAndSave',
+            src: (context) => {
+              return scanAndSave(context.driver);
+            },
+            onDone: {
+              target: 'interpreting',
+              actions: assign({ scannedImagePaths: (_, event) => event.data }),
+            },
           },
-          onDone: {
-            target: 'interpreting',
-            actions: assign({ scannedImagePaths: (_, event) => event.data }),
+          pollPaperStatus(),
+        ],
+        on: {
+          PAPER_JAM: {
+            target: 'jammed',
           },
         },
       },
       interpreting: {
+        // Paper is in the paper handler for the duration of the interpreting stage and paper handler
+        // motors are never moved, so we don't need to poll paper status or handle jams.
         invoke: {
           id: 'interpretScannedBallot',
           src: loadMetadataAndInterpretBallot,
           onDone: {
             target: 'presenting_ballot',
+            actions: assign({
+              interpretation: (_, event) => {
+                debug('Interpretation complete; storing to context');
+                return event.data;
+              },
+            }),
           },
         },
       },
       presenting_ballot: {
+        invoke: pollPaperStatus(),
+        entry: (context) => {
+          debug('+entry for presenting_ballot');
+          if (context.interpretation) {
+            // `debug`'s string interpolation only pretty prints 1 layer deep; we need 2
+            debug(
+              `Front page interpretation:\n${JSON.stringify(
+                context.interpretation[0].interpretation,
+                null,
+                2
+              )}`
+            );
+            debug(
+              `Back page interpretation:\n${JSON.stringify(
+                context.interpretation[1].interpretation,
+                null,
+                2
+              )}`
+            );
+          } else {
+            debug('No interpretation found in context');
+          }
+          debug('Presenting paper');
+          void context.driver.presentPaper();
```

Comment:
Hmm I thought I read something about entry actions not supporting async/await syntax but I think I misunderstood because I can't find it in the docs. [04a8f72](https://github.com/votingworks/vxsuite/pull/3855/commits/04a8f72588ee7bd01e18b6f8c0b850f694eb26ec)